### ISSUE TITLE
arbitrary-image-stylization-v1-256

### ISSUE THREAD

**Issue by Jonathhhan** [zkmkarlsruhe/ofxTensorFlow2#21]
I try to run the arbitrary-image-stylization-v1-256 model: https://tfhub.dev/google/magenta/arbitrary-image-stylization-v1-256/2
I can load the model without any errors, but it chrashes if I try to run the model (without an error message).
This is how I try to run it:
```
	if(!model.load("magenta")) {
		std::exit(EXIT_FAILURE);
	}

	model.setup({ {"serving_default_placeholder:0"} ,{"serving_default_placeholder_1:0"} }, { {"StatefulPartitionedCall:0"} });

	auto input_1 = cppflow::decode_jpeg(cppflow::read_file(std::string(ofToDataPath("example_1.jpg"))));
	input_1 = cppflow::cast(input_1, TF_UINT8, TF_FLOAT);
	input_1 = cppflow::expand_dims(input_1, 0);
	auto input_2 = cppflow::decode_jpeg(cppflow::read_file(std::string(ofToDataPath("example_2.jpg"))));
	input_2 = cppflow::cast(input_2, TF_UINT8, TF_FLOAT);
	input_2 = cppflow::expand_dims(input_2, 0);
	std::vector<cppflow::tensor> vectorOfInputTensors = {
	 input_1, input_2
	};
	auto output = model.runModel(vectorOfInputTensors);
```
Any idea what I am doing wrong (or if the model runs at all with ofxTensorflow2)? Thank you.

**Comment by bytosaur** [comment]
Hey @Jonathhhan,

I am currently on vacation and can only look into it at the mid of next week.
It looks alright how you handle the input. Maybe you are running out of Gpu memory? Try setting memory growth with ofxTF2::setGPUmem(90, true)

**Comment by Jonathhhan** [comment]
Hey @bytosaur,
I think I found a solution and made an example: https://github.com/Jonathhhan/ofxTensorFlow2/tree/universal-sentence-encoder/example_arbitrary-image-stylization
The advantage is (in my opinion) that you just need to load an image and not a model for the style.

**Comment by bytosaur** [comment]
Yup arbitrary style transfer is superior in that sense. How much fps do you get and what hardware do you use? 
The same model has been used in Alexander Schubert's crawl3rs ;)

**Comment by Jonathhhan** [comment]
It runs at 60FPS with a RTX3090, but it needs a lot of power (around 340w). Not sure how to reduce that.

**Comment by bytosaur** [comment]
hey @Jonathhhan,

thank you for your contribution! I am currently trying out your example on an RTX 3080.
I have two ideas how to improve the performance:
- use multiply by 1/255.0 instead of div by 255
	```c++
	input2 = cppflow::mul(input2, cppflow::tensor({ 1.0f/255.f}));
	```
- allocate the image beforehand instead of for every image
	```c++
	imgOut.allocate(videoPlayer.getWidth(), videoPlayer.getHeight(), OF_IMAGE_COLOR);
	```

Please feel free to make a PR for this example!

BTW could you please name the example example_style_transfer_arbitrary? This way they are next to each other :) and i would like to keep the basic example on top

**Comment by bytosaur** [comment]
oh and please keep in mind to use the standard OF_ROOT. I.e. leave the variable commented in config.make ;)

**Comment by Jonathhhan** [comment]
hey @bytosaur,
thanks for the improvement ideas. mul instead of div does not seem to make a noticable difference and i need to allocate the output image after running the model, because sometimes the output shape differs slightly from the input (in my case the width difference is 2px). 
Here is a branch with just the example. I changed the name and made some small changes:
https://github.com/Jonathhhan/ofxTensorFlow2/tree/example_style_transfer_arbitrary/example_style_transfer_arbitrary
I also tried to adapt the example_style_transfer example but I think I had problems to setup the model in the model class, so it didnt work (but in theory it should ;)).
And I copied config.make from the other example, cant remember that I changed something (also not sure how it works, exactly).

**Comment by Jonathhhan** [comment]
Alright :) It seemed that the video had a strange resolution or was corrupted, with another video it works without allocating from the shape. I also replaced div with mul. And I read the video texture directly into float pixels now, so now further conversion is needed. I will have a look into the other things...

**Comment by bytosaur** [comment]
hey @Jonathhhan,

i have already pushed a branch with some smaller changes. I ll first adjust and merge your other examples before modifying the download scripts :)

**Comment by Jonathhhan** [comment]
hey @bytosaur,
thanks a lot for improving and polishing the examples (and the addon, of course). I am still a beginner and already learned something from it.
I also found some small things that I could have done better (but nothing with a big increase in performance).
I can do some pull requests from your branches.
This way for example, the ofFloatPixels object is not needed anymore:

```
		input = ofxTF2::pixelsToTensor(videoPlayer.getPixelsRef());
		input = cppflow::expand_dims(input, 0);
		input = cppflow::resize_bicubic(input, cppflow::tensor({ 360, 480 }), true);
		input = cppflow::cast(input, TF_UINT8, TF_FLOAT);
		input = cppflow::div(input, cppflow::tensor({ 255.f }));
```

**Comment by bytosaur** [comment]
hey @Jonathhhan,
we highly appreciate your examples! I didnt do much on the code side of things except for packing the inference part into a separate function. Your examples have been pushed to [this branch](https://github.com/zkmkarlsruhe/ofxTensorFlow2/tree/feature/new_examples) and will be merged soon :)

**Comment by bytosaur** [comment]
included in #29
